### PR TITLE
VSTestPlusPlus with UI

### PR DESCRIPTION
**Pull request by RanjanarMS**

### CODE REVIEW

File: Tasks/VsTest/task.json [hunk 1]
```diff
@@ -1,198 +1,260 @@
...
+	"runsOn": [
         "Agent",
         "MachineGroup"
     ],
     "author": "Microsoft Corporation",
     "version": {
         "Major": 2,
         "Minor": 0,
-        "Patch": 11
+        "Patch": 100
```
Comment (acesiddhu) [inline Tasks/VsTest/task.json]: >100 [](start = 17, length = 3)

any reason why 100?

File: Tasks/VsTest/task.json [hunk 2]
```diff
@@ -1,198 +1,260 @@
...
     },
     "demands": [
         "vstest"
     ],
     "preview": "true",
-    "releaseNotes": "<ul><li><b>Test Impact Analysis:</b> Automatically select and run only the tests needed to validate the code change.</li><li>Support for minimatch patterns for specifying test assemblies.</li></ul>",
+    "releaseNotes" : "Adding the functionality to run distributed test with release phases.",
```
Comment (acesiddhu) [inline Tasks/VsTest/task.json]: >releaseNotes" : "Adding the functionality to run distributed test with release phases [](start = 5, length = 85)

why remove old release notes? can you check with Prachi here

File: Tasks/VsTest/task.json [hunk 3]
```diff
@@ -1,198 +1,260 @@
 {
     "id": "EF087383-EE5E-42C7-9A53-AB56C98420F9",
     "name": "VSTest",
-    "friendlyName": "Visual Studio Test",
+    "friendlyName": "Visual Studio Test Plus Plus",
```
Comment (acesiddhu) [inline Tasks/VsTest/task.json]: >Plus Plus [](start = 40, length = 9)

why change this?

File: Tasks/VsTest/task.json [hunk 4]
```diff
@@ -1,198 +1,260 @@
...
     "visibility": [
         "Build",
         "Release"
     ],
-    "runsOn": [
+	"runsOn": [
         "Agent",
         "MachineGroup"
     ],
     "author": "Microsoft Corporation",
```
Comment (acesiddhu) [inline Tasks/VsTest/task.json]: can we test upgrade scenario by creating a build definition with 2.11 task and then uploading this task and check if we queue build things are fine

File: Tasks/VsTest/task.json [hunk 5]
```diff
@@ -1,198 +1,260 @@
...
+            "required": false,  
+            "helpMarkDown": "The version of Visual Studio test to use.",  
+            "groupName": "executionOptions",  
+            "options": {  
+                 "15.0": "Visual Studio 2017",  
+                 "14.0": "Visual Studio 2015"  
+             }  
+        },         
         {
-            "name": "vstestLocationMethod",
```
Comment (acesiddhu) [inline Tasks/VsTest/task.json]: is there a way we can make sure thta definition which have selected this option can fall back to latest option automatically. so as we dont break their builds?

File: Tasks/VsTest/task.json [hunk 6]
```diff
@@ -1,198 +1,260 @@
...
             "defaultValue": "50",
             "required": false,
             "helpMarkDown": "Number of builds after which to automatically run all tests. Test Impact Analysis stores the mapping between test cases and source code. It is recommended to regenerate the mapping by running all tests, on a regular basis.",
-            "groupName": "tiaOptions"
-        },
+            "groupName": "testSelection",
+            "visibleRule": "testSelector = testAssemblies && runOnlyImpactedTests = true"
+        },      
+        {  
+           "name": "testPlatform",
```
Comment (acesiddhu) [inline Tasks/VsTest/task.json]: >testPlatform [](start = 20, length = 12)

why change the name? wont it break existing build def?

File: Tasks/VsTest/distributedTest.ts [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+            }
+            tl.debug(fs.readFileSync(this.dtaHostLogFilePath, 'utf-8'));
+            tl.setResult(tl.TaskResult.Succeeded, 'Task succeeded');
+        } catch (error) {
+            tl.debug('Create Agent failed with  : ' + error);
+            tl.setResult(tl.TaskResult.Failed, 'Task failed');
+        }
+    }
+
+    private async createAgent(retries: number) {
```
Comment (acesiddhu) [inline Tasks/VsTest/distributedTest.ts]: should we refactor agent command and run command in separate classes / files?

File: Tasks/VsTest/task.json (hunk 1 above)
Comment (RanjanarMS) [inline Tasks/VsTest/task.json]: Removed

File: Tasks/VsTest/task.json (hunk 2 above)
Comment (RanjanarMS) [inline Tasks/VsTest/task.json]: Will sync up with Prachi and put the right text in next iteration.

File: Tasks/VsTest/task.json (hunk 3 above)
Comment (RanjanarMS) [inline Tasks/VsTest/task.json]: Removed

File: Tasks/VsTest/distributedTest.ts (hunk 7 above)
Comment (RanjanarMS) [inline Tasks/VsTest/distributedTest.ts]: Moved TestAgent out.. And helper functions out

File: Tasks/VsTest/task.json (hunk 1 above)
Comment (acesiddhu) [inline Tasks/VsTest/task.json]: >100 [](start = 17, length = 3)

i still see this as 100

File: Tasks/VsTest/task.json [hunk 8]
```diff
@@ -1,198 +1,260 @@
...
-            "defaultValue": "version",
-            "options": {
-                "version": "Version",
-                "location": "Specify Location"
-            }
+            "helpMarkDown": "Path to runsettings file to use with the tests. Use `$(Build.SourcesDirectory)` to access the Project folder.",
+            "groupName": "executionOptions"
         },
         {
-            "name": "vsTestVersion",
```
Comment (acesiddhu) [inline Tasks/VsTest/task.json]: >vsTestVersion [](start = 21, length = 13)

if we rename this option people will always get vs 2015. specially when they are using test impact using 2017

**Comment by acesiddhu** [comment]
>         "Patch": 11

I am assuming all the change in master before it is pushed we will have bumped up the patch version. else next sprint deployment wont contains these changes.

---
Refers to: Tasks/VsTest/task.json:20 in ea1559f. [](commit_id = ea1559f8db6d17c3e3e2e8d3fbca114777a30931, deletion_comment = False)

**Comment by acesiddhu** [comment]
:shipit:

**Pull request merged**